p2p/src/sync/peer.rs

```diff
@@ -432,11 +432,20 @@ where
         let block = self.chainstate_handle.call(|c| c.preliminary_block_check(block)).await??;
         match self
             .chainstate_handle
-            .call_mut(|c| c.process_block(block, BlockSource::Peer))
+            .call_mut(|c| {
+                // If the block has already been processed, return the existing block index.
+                // It's used to prevent chainstate from printing "Block already exists" errors.
+                if let Some(block_index) = c.get_block_index(&block.get_id())? {
+                    log::debug!("Block ({}) already exists", block.get_id());
```

Comment:
It's a good idea to add the context of this message, like "peer X sent a block that already exists". This helps in fine-tuning the process later.